### PR TITLE
Rename CDC terms

### PR DESCRIPTION
**Pull request by erickpintor**
### Description

This patch renames CDC terms according to our new guidelines.

### Motivation and context

* "Steam token" becomes "Event source"
* `.change_feed(..)` becomes `.feed(..)`
* `ChangeFeed{Options,Iterator,Page}` becomes `Feed{Options,Iterator,Page}`

Old terms related to streaming are deprecated and should be removed in the next major version. Old terms related to event feeds were ripped off.

### How was the change tested?

Unit tests included.

### Screenshots (if appropriate):

NA.

### Change types

* - [ ] Bug fix (non-breaking change that fixes an issue)
* - [X] New feature (non-breaking change that adds functionality)
* - [ ] Breaking change (backwards-incompatible fix or feature)

### Checklist:

* - [X] My code follows the code style of this project.
* - [X] My change requires a change to Fauna documentation.
* - [X] My change requires a change to the README, and I have updated it accordingly.

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by erickpintor** [comment]
FYI: This is ready for a code review but tests will fail until the next docker release is out some time later today.

File: fauna/client/client.py [hunk 1]
```diff
@@ -420,13 +419,13 @@ def _query(
...
       self,
-      fql: Union[StreamToken, Query],
+      fql: Union[EventSource, Query],
       opts: StreamOptions = StreamOptions()
   ) -> "StreamIterator":
     """
         Opens a Stream in Fauna and returns an iterator that consume Fauna events.
 
-        :param fql: A Query that returns a StreamToken or a StreamToken.
+        :param fql: A Query that returns a EventSource or a EventSource.
```
Comment (cynicaljoy) [inline fauna/client/client.py:428]: consider using a different arg value and changing the wording

File: fauna/client/client.py [hunk 2]
```diff
@@ -459,18 +458,18 @@ def stream(
...
-      opts: ChangeFeedOptions = ChangeFeedOptions()
-  ) -> "ChangeFeedIterator":
+      fql: Union[EventSource, Query],
+      opts: FeedOptions = FeedOptions(),
+  ) -> "FeedIterator":
     """
         Opens an Event Feed in Fauna and returns an iterator that consume Fauna events.
 
-        :param fql: A Query that returns a StreamToken or a StreamToken.
+        :param fql: A Query that returns a EventSource or a EventSource.
```
Comment (cynicaljoy) [inline fauna/client/client.py:469]: see previous comment

**Review by jrodewig — COMMENTED** [review]
Look good overall! Left some minor nits and feedback.

**Comment by findgriffin** [comment]
Not sure if it fits in this PR, but something I just came across while doing event feeds for JVM is that we have a lot of classes with _Query_ in the name that we might want to rename. Things like `QueryFailure`, `QueryTimeout`.

File: fauna/client/client.py [hunk 3]
```diff
@@ -459,18 +458,18 @@ def stream(
     return StreamIterator(self._session, headers, self._endpoint + "/stream/1",
                           self._max_attempts, self._max_backoff, opts, token)
 
-  def change_feed(
+  def feed(
       self,
-      fql: Union[StreamToken, Query],
-      opts: ChangeFeedOptions = ChangeFeedOptions()
-  ) -> "ChangeFeedIterator":
+      fql: Union[EventSource, Query],
```
[the remark this answers is not in the document]
Comment (erickpintor) [reply fauna/client/client.py:463]: This one we can break. The one on the `def stream(..)` function we can't without a major release.

**Comment by erickpintor** [comment]
> Not sure if it fits in this PR, but something I just came across while doing event feeds for JVM is that we have a lot of classes with Query in the name that we might want to rename. Things like QueryFailure, QueryTimeout.

I'd consider that a separate effort.

**Comment by erickpintor** [comment]
This is ready for another pass.

File: README.md [hunk 4]
```diff
@@ -292,13 +292,12 @@ The driver supports [Event Feeds](https://docs.fauna.com/fauna/current/learn/tra
 ### Request an Event Feed
 
 An Event Feed asynchronously polls an [event stream](https://docs.fauna.com/fauna/current/learn/streaming),
-represented by a stream token, for events.
+represented by an event source, for events.
```
Comment (jrodewig) [inline README.md:295]:
```
An Event Feed asynchronously polls an [event source](https://docs.fauna.com/fauna/current/learn/streaming),
for paginated events.
```

**Review by jrodewig — DISMISSED** [review]
LGTM aside from a minor fix in the README. Thanks!

File: README.md (hunk 4 above)
Comment (erickpintor) [reply README.md]: Done. Thank you!

**Review by pnwpedro — APPROVED** [review]

**Pull request merged**